### ISSUE TITLE
Questions regarding training conformer transducer model.

### ISSUE THREAD

**Issue by gabitza-tech** [NVIDIA/NeMo#6990]
Hello,

Thank you for your work and all the help provided so far with other issues too. 

I am training a conformer transducer model and I have a couple of questions that might help me achieve better results:

1. I am finetuning the **stt en conformer transducer medium** model (whole model) on 225h of romanian language ( 200h training, 20h validation). I am at epoch 100 and the val_wer keeps decreasing, as well as the training wer. However, the performances on the eval set are still far away in performance comparing to our Kaldi model (10-15% more WER). How many epochs are usually necessary for convergence of the transducer medium model and are 200 hours of training data enough for a conformer model?

2. Do you use any type of cross-validation or simply randomly split the training set in train/val? 

3. At the moment I am using specaugment and spec cutout, should I also add noise impulses/background noise from some datasets too as augmentation? Could you please point what I should add in the config file exactly or if there is an example I could follow?

4. Lastly, I am using an LR of 1 with Noam annealing and 10% warm up ratio, is it a big LR for finetuning? ( from what I understood, you used 5 as LR) I am unsure if I should lower it to 0.1 maybe or 0.01.

Training takes quite a long time on our setup and we can't afford to test many configurations, that's why these questions may be pretty basic, but any insight is greatly appreciated!

**Comment by VahidooX** [comment]
>>>How many epochs are usually necessary for convergence of the transducer medium model and are 200 hours of training data enough for a conformer model?
RNNT models converge significantly faster than CTC models. For instance on LS, the large model converges in around 200 epochs.
You need more epochs. But the slow convergence is probably due to low learning rate and high augmentation. 
On data size, more data always helps. 

>>>Do you use any type of cross-validation or simply randomly split the training set in train/val?
Most of the speech datasets have been already split into train/validation sets. You may just split your data randomly. Maybe 20 hours for validation and the rest to be used for training. 

>>>At the moment I am using specaugment and spec cutout, should I also add noise impulses/background noise from some datasets too as augmentation? Could you please point what I should add in the config file exactly or if there is an example I could follow? Adding more noise augmentation would slow down your training even more. However it can help with the accuracy of the final model if you need to be working in noisy environments. I would update here with more detail on how to do noise augmentation later.

>>>Lastly, I am using an LR of 1 with Noam annealing and 10% warm up ratio, is it a big LR for finetuning? ( from what I understood, you used 5 as LR) I am unsure if I should lower it to 0.1 maybe or 0.01.
You are fine-tuning on another language. In such cases, maybe you can even try larger lr like 2 or 5. Specially if you want to converge faster. If you have limited time and resource, always start new experiments from the last checkpoint you trained to save some time. Not sure if cutoff augmentation is needed when you use specaug.

**Comment by gabitza-tech** [comment]
Hi @VahidooX!

Thank you for your comprehensive answer! It helped a lot and maanged to train some models pretty well. I also managed to perform speed perturbations for augmentations, but when trying to perform noise augmentation, I get "??" in predictions..

Do you know what may cause this behavior? I described this issue in more detail here: https://github.com/NVIDIA/NeMo/issues/7145

Best regards,
Gabi
